maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/util/DependencyScanner.java

```diff
@@ -111,19 +112,46 @@ private static void scanArtifact( File artifact, TestFilter<String, String> filt
         {
             for ( String groups : groupArtifactIds )
             {
+                // groupId:artifactId[:version[:type[:classifier]]]
                 String[] groupArtifact = groups.split( ":" );
-                if ( groupArtifact.length != 2 )
+                if ( groupArtifact.length < 2 || groupArtifact.length > 5 )
                 {
                     throw new IllegalArgumentException( "dependencyToScan argument should be in format"
-                        + " 'groupid:artifactid': " + groups );
+                        + " 'groupid:artifactid[:version[:type[:classifier]]]': " + groups );
                 }
-                if ( artifact.getGroupId().matches( groupArtifact[0] )
-                    && artifact.getArtifactId().matches( groupArtifact[1] ) )
+                if ( artifactMatchesGavtc( artifact, groupArtifact ) )
                 {
                     matches.add( artifact.getFile() );
                 }
             }
         }
         return matches;
     }
+    
+    private static boolean artifactMatchesGavtc( Artifact artifact, String[] gavtc )
```

Comment:
In my next comment I use "groups" as a parameter because the upper part has local variable "groups". You can rename it as you like.